dask/array/core.py

```diff
@@ -1012,9 +1011,9 @@ def broadcast_chunks(*chunkss):
 
 def store(
     sources: Array | Collection[Array],
-    targets,
+    targets: ArrayLike | Delayed | Collection[ArrayLike | Delayed],
     lock: bool | Lock = True,
-    regions: tuple[slice, ...] | Collection[tuple[slice, ...]] | None = None,
+    regions: tuple[slice, ...] | Collection[tuple[slice, ...] | None] | None = None,
```

Comment:
This one is interesting. The `None` that I added is inside the `Collection[...]`, meaning (as I see it) a sequence of Nones. When I take this out, mypy complains about this line:

https://github.com/dask/dask/blob/d7a64d723134b9c51e0e38cbe0a94f13349ffddd/dask/array/core.py#L1094-L1095

Reply:
This is ugly. In order to make the implementation work, you made the input spec more lax. This could be avoided by just using a different variable name.